cloudquery/sdk/scalar/scalar_factory.py

```diff
@@ -22,74 +24,39 @@ def __init__(self):
-        #     return ()
-        # elif dt_id == pa.types.lib.Type_DURATION:
-        #     return ()
-        elif dt_id == pa.types.lib.Type_DOUBLE:
-            return Float(bitwidth=64)
-        elif dt_id == pa.types.lib.Type_FLOAT:
-            return Float(bitwidth=32)
-        elif dt_id == pa.types.lib.Type_HALF_FLOAT:
-            return Float(bitwidth=16)
-        # elif dt_id == pa.types.lib.Type_INTERVAL_MONTH_DAY_NANO:
-        #     return ()
-        elif (
-            dt_id == pa.types.lib.Type_LIST
-            or dt_id == pa.types.lib.Type_LARGE_LIST
-            or dt_id == pa.types.lib.Type_FIXED_SIZE_LIST
-        ):
-            item = ScalarFactory.new_scalar(dt.field(0).type)
-            return List(type(item))
-        # elif dt_id == pa.types.lib.Type_MAP:
-        #     return ()
-        elif (
-            dt_id == pa.types.lib.Type_STRING or dt_id == pa.types.lib.Type_LARGE_STRING
-        ):
-            return String()
-        # elif dt_id == pa.types.lib.Type_STRUCT:
-        #     return ()
-        # elif dt_id == pa.types.lib.Type_TIME32:
-        #     return ()
-        # elif dt_id == pa.types.lib.Type_TIME64:
-        #     return ()
-        elif dt_id == pa.types.lib.Type_TIMESTAMP:
-            return Timestamp()
-        elif dt == UUIDType():
+        type_map = {
+            pa.types.lib.Type_INT64: partial(Int, bitwidth=64),
+            pa.types.lib.Type_INT32: partial(Int, bitwidth=32),
+            pa.types.lib.Type_INT16: partial(Int, bitwidth=16),
+            pa.types.lib.Type_INT8: partial(Int, bitwidth=8),
+            pa.types.lib.Type_UINT64: partial(Uint, bitwidth=64),
+            pa.types.lib.Type_UINT32: partial(Uint, bitwidth=32),
+            pa.types.lib.Type_UINT16: partial(Uint, bitwidth=16),
+            pa.types.lib.Type_UINT8: partial(Uint, bitwidth=8),
+            pa.types.lib.Type_BINARY: Binary,
+            pa.types.lib.Type_LARGE_BINARY: Binary,
+            pa.types.lib.Type_FIXED_SIZE_BINARY: Binary,
+            pa.types.lib.Type_BOOL: Bool,
+            pa.types.lib.Type_DATE64: Date64,
+            pa.types.lib.Type_DATE32: Date32,
+            pa.types.lib.Type_DOUBLE: partial(Float, bitwidth=64),
+            pa.types.lib.Type_FLOAT: partial(Float, bitwidth=32),
+            pa.types.lib.Type_HALF_FLOAT: partial(Float, bitwidth=16),
+            pa.types.lib.Type_LIST: List,
+            pa.types.lib.Type_LARGE_LIST: List,
+            pa.types.lib.Type_FIXED_SIZE_LIST: List,
+            pa.types.lib.Type_STRING: String,
+            pa.types.lib.Type_LARGE_STRING: String,
+            pa.types.lib.Type_TIMESTAMP: Timestamp,
+        }
+        if dt_id in type_map:
+            scalar_type = type_map[dt_id]
+            if scalar_type == List:
+                item = self.new_scalar(dt.field(0).type)
+                return scalar_type(type(item))
+            return scalar_type()
+        if dt == UUIDType():
```

Comment:
add `UUIDType` to `type_map`?

Reply:
So, in this case, you'll notice we're not comparing `dt_id` but `dt` instead.
I didn't check why this is the case in the original code and simply kept the logic the same in this change. 

If this was a mistake in the original code, and it should check against `dt_id`, then we can absolutely add it to the `type_map`. 
If, however, this was the intended logic, then there is a risk that by adding it to the `type_map` and implementing a separate check for `dt` against the `type_map` we introduce some unexpected bug.

Reply:
@hermanschaaf to the rescue